### PR TITLE
Update ubuntu version in template-engine.yml

### PR DESCRIPTION
**Pull request by v-wuzhai**

### CODE REVIEW

**Comment by v-wuzhai** [comment]
Switched to Microsoft-hosted agent pool because of image 1es-Ubuntu-2204-open doesn't exist in pool NetCore-Svc-Public

File: eng/template-engine.yml [hunk 1]
```diff
@@ -24,18 +24,17 @@ jobs:
...
-            name: NetCore-Svc-Public
-            demands: ImageOverride -equals 1es-ubuntu-2004-open
+            vmImage: 'ubuntu-22.04'
           ${{ if ne(variables['System.TeamProject'], 'public') }}:
             name: NetCore1ESPool-Svc-Internal
-            demands: ImageOverride -equals 1es-ubuntu-2004
+            demands: ImageOverride -equals 1es-ubuntu-2204
         ${{ if eq(variables['System.TeamProject'], 'public') }}:
-          helixTargetQueue: 'ubuntu.2004.amd64.open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-20.04-helix-amd64-20220502145738-4b2e4c2'
+          helixTargetQueue: 'ubuntu.2204.amd64.open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-22.04-helix-amd64-20220813234317-1b9461f'
```
Comment (kasperk81) [inline eng/template-engine.yml:35]: use the [latest tag](https://github.com/dotnet/versions/blob/main/build-info/docker/image-info.dotnet-dotnet-buildtools-prereqs-docker-main.json)?
```suggestion
          helixTargetQueue: 'ubuntu.2204.amd64.open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-22.04-helix-amd64-20220915134500-1b9461f'
```

File: eng/template-engine.yml [hunk 2]
```diff
@@ -24,18 +24,17 @@ jobs:
...
           ${{ if ne(variables['System.TeamProject'], 'public') }}:
             name: NetCore1ESPool-Svc-Internal
-            demands: ImageOverride -equals 1es-ubuntu-2004
+            demands: ImageOverride -equals 1es-ubuntu-2204
         ${{ if eq(variables['System.TeamProject'], 'public') }}:
-          helixTargetQueue: 'ubuntu.2004.amd64.open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-20.04-helix-amd64-20220502145738-4b2e4c2'
+          helixTargetQueue: 'ubuntu.2204.amd64.open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-22.04-helix-amd64-20220915134500-1b9461f'
         ${{ if ne(variables['System.TeamProject'], 'public') }}:
-          helixTargetQueue: Ubuntu.2004.Amd64
+          helixTargetQueue: Ubuntu.2204.Amd64
```
Comment (marcpopMSFT) [inline eng/template-engine.yml:37]: we should hold off on this. We discovered that 2204 is not yet available in the internal helix queues so even though this passes in the public PR, the matching internal PRs fail.

**Comment by marcpopMSFT** [comment]
Since this is for rc2 and internal builds don't support 2204 yet, closing this out for now. Once we have that, we can open a 7.0.1xx pr.

**Pull request closed without merging**